### PR TITLE
fix: Navbar Close on link click #2464

### LINKED ISSUE #2464: Navbar Close on link click
### Description

We can close the navbar ( mobile view ) When the routes are changed or A link is clicked

image

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by talalanjum**
## Fixes Issue
Closes #2464

## Changes proposed

We can close the navbar ( mobile view ) when the routes are changed or a link is clicked

## Check List (Check all the applicable boxes) 

- [X] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] All new and existing tests passed.
- [X] This PR does not contain plagiarized content.
- [X] The title of my pull request is a short description of the requested changes.

## Screenshots

image

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/2507"></a>

### CODE REVIEW

**Review by Aadarsh805 — COMMENTED** [review]
This works fine, but this also applies for desktop screen right
a better solution would have been setting isOpen to false on every navlink click like this
```
<Link href={item.url} className={className} onClick={() => setIsOpen(false)} aria-current="page">
      {item.name}
</Link>
```
What do you think

**Comment by talalanjum** [comment]
> This works fine, but this also applies for desktop screen right a better solution would have been setting isOpen to false on every navlink click like this
> 
> ```
> <Link href={item.url} className={className} onClick={() => setIsOpen(false)} aria-current="page">
>       {item.name}
> </Link>
> ```
> 
> What do you think

**Comment by talalanjum** [comment]
> This works fine, but this also applies for desktop screen right a better solution would have been setting isOpen to false on every navlink click like this
> 
> ```
> <Link href={item.url} className={className} onClick={() => setIsOpen(false)} aria-current="page">
>       {item.name}
> </Link>
> ```
> 
> What do you think

It does seem to be a better solution as it won't run the close process on desktop nav clicks. Made the changes for you to re-review

File: components/navbar/Navbar.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { useRouter } from "next/router";
-import { useState } from "react";
+import { useState, useEffect } from "react";
```
Comment (ManishBisht777) [inline components/navbar/Navbar.js:2]: Does This `useEffect` has some use?

Comment (talalanjum) [reply components/navbar/Navbar.js]: typo, fixed

**Comment by ChinmayMhatre** [comment]
Hey, I looked into the pr. Looks good.

**Review by ManishBisht777 — COMMENTED** [review]
looks good to me

**Comment by Aadarsh805** [comment]
I tried running it and it isn't working 
It's giving `setIsOpen is not a function error`

**Comment by talalanjum** [comment]
> I tried running it and it isn't working It's giving `setIsOpen is not a function error`

Please check if you have the latest changes, and if the problem still persists, can you show me the steps to reproduce?

**Comment by ChinmayMhatre** [comment]
> > I tried running it and it isn't working It's giving `setIsOpen is not a function error`
> 
> Please check if you have the latest changes, and if the problem still persists, can you show me the steps to reproduce?

@Aadarsh805  have you tried this?

**Comment by Aadarsh805** [comment]
> 



> > > I tried running it and it isn't working It's giving `setIsOpen is not a function error`
> > 
> > 
> > Please check if you have the latest changes, and if the problem still persists, can you show me the steps to reproduce?
> 
> @Aadarsh805 have you tried this?

yeah that error doesn't appear anymore, must have been because of not having the latest changes

**Comment by ChinmayMhatre** [comment]
@eddiejaoude could you take a look.

**Comment by eddiejaoude** [comment]
Thank you for the reminder @ChinmayMhatre 👍 I will try to look at it soon

**Review by eddiejaoude — APPROVED** [review]
Looks great 👍  thank you 🔥

**Pull request merged**